Resources/Prototypes/Floof/Entities/Objects/Weapons/guns.yml

```diff
@@ -0,0 +1,89 @@
+
+- type: entity
+  name: borg disabler
+  parent: WeaponDisabler
+  id: WeaponBorgDisabler
+  description: A self-defense weapon that exhausts organic targets, weakening them until they collapse.
+  components:
+    - type: BatterySelfRecharger
+      autoRecharge: true
+      autoRechargeRate: 30
+
+# - type: entity
+  # name: borg taser
+  # parent: BaseWeaponBatterySmall
+  # id: WeaponborgTaser
+  # description: A low-capacity, energy-based stun gun used by security teams to subdue targets at range.
+  # components:
+  # - type: Tag
+    # tags:
+    # - Taser
+    # - Sidearm
+  # - type: Sprite
+    # sprite: Objects/Weapons/Guns/Battery/taser.rsi
+    # layers:
+    # - state: base
+      # map: ["enum.GunVisualLayers.Base"]
+    # - state: mag-unshaded-0
+      # map: ["enum.GunVisualLayers.MagUnshaded"]
+      # shader: unshaded
+  # - type: Item
+    # heldPrefix: taser4
+  # - type: Clothing
+    # quickEquip: false
+    # slots:
+    # - Belt
+  # - type: Gun
+    # soundGunshot:
+      # path: /Audio/Weapons/Guns/Gunshots/taser.ogg
+  # - type: ProjectileBatteryAmmoProvider
+    # proto: BulletTaser
+    # fireCost: 200
+  # - type: BatterySelfRecharger
+    # autoRecharge: true
+    # autoRechargeRate: 40
+  # - type: MagazineVisuals
+    # magState: mag
+    # steps: 5
+    # zeroVisible: true
+  # - type: Appearance
+
+- type: entity
+  name: Borg mk 58
+  parent: BaseWeaponBatterySmall
+  id: WeaponborgPistolMk58
+  description: This is an antique laser pistol. All craftsmanship is of the highest quality. It is decorated with assistant leather and chrome. The object menaces with spikes of energy.
```

Comment:
That is... not an mk-58. According to the components provided, it's an antique laser pistol that fires... 10 PKA bullets at the rate of 2 bullets per second?

Reply:
Dont ingore those comments and just resolve it, this is correct your shotting Shuttle PKA "BulletKineticShuttle" instead of BulletPistol.

Suggested changes below.

Reply:
This was discussed privately and explained, you know that wouldn't work and why, why are you making suggestions that would make the code throw errors or not work?

Reply:
> This was discussed privately and explained, you know that wouldn't work and why, why are you making suggestions that would make the code throw errors or not work?

The code i suggested was TESTED AND WORKS.

Reply:
you tested this and got it working? alright.